projects/baby_driver/scripts/test_adc_read.py

```diff
@@ -0,0 +1,49 @@
+"""This Module Tests methods in adc_read.py"""
+import unittest
+from unittest.mock import patch
+
+from adc_read import adc_read, NUM_PINS_PER_PORT
+from can_util import Message
+from gpio_port import GpioPort
+
+RAW_MAX = 4095
+CONVERTED_MAX = 3000
+
+class TestADCRead(unittest.TestCase):
+    """Test Babydriver's adc_read function"""
+
+    @patch('can_util.send_message')
+    @patch('can_util.next_message')
+    def test_check_range_pin(self):
+        """Tests Maximum values"""
+        self.assertLessEqual(adc_read(GpioPort.A, 0, True), RAW_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 1, True), RAW_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 2, True), RAW_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 0, False), CONVERTED_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 1, False), CONVERTED_MAX)
+        self.assertLessEqual(adc_read(GpioPort.A, 2, False), CONVERTED_MAX)
```

Comment:
This test seems to be testing properties of the ADC implementation, rather than the Python interface. I think a better test of the Python would be to use the mocks to set the return value of `next_message` to a Message that matches CAN messages the C should provide, then check that `adc_read` correctly parses and returns the values from it.

You should also have another test that uses the `side_effect` of `mock_send_message` to intercept the data being sent and ensures that it's in the correct format.